### ISSUE TITLE
Request for Docker Image for CLAIRE 

### ISSUE THREAD

**Issue by LalithShiyam** [andreasmang/claire#16]
Hi CLAIRE Team,

I hope this message finds you well.

First, thank you for your fantastic work on CLAIRE. I read your manuscript with great interest and I’m genuinely excited to use it in our research with total-body PET. It's not every day someone casually drops a diffeomorphic tool, with heavy parallelization/optimization in mind. 

However, I’ve been struggling with the installation process. Despite following the instructions carefully, I’ve run into several issues that I haven’t been able to resolve. This has made it difficult for me to get CLAIRE up and running.

I believe an official Docker image for CLAIRE could make the installation much more straightforward. It would help not only me but also others who might be facing similar challenges. I considered creating a Docker image myself to contribute back, but my difficulties with the initial setup have prevented me from doing so.

Would you consider providing a Docker image for CLAIRE? I think it would be a great help to many in the community. If there’s any way I can assist in this process, please let me know.

Thank you for your time and consideration.

Cheers,
Lalith

**Comment by naveenaero** [comment]
Hi LalithShiyam,
Happy to hear that CLAIRE could be of some use to you. Could you please let us know your use case and environment you will be working on so we would better help you?
Specifically if you could let us know whether you will use single-GPU vs multi-gpu or just multi-core CPU?

**Comment by LalithShiyam** [comment]
Hi @naveenaero,

Thank you for your prompt reply.

We routinely perform motion correction in 4D whole-body PET images using the [Greedy](https://github.com/pyushkevich/greedy) toolkit. We've developed a wrapper to customize it for our [specific use case](https://github.com/ENHANCE-PET/FALCON). Although Greedy operates solely on CPU and performs alignments efficiently using multi resolution paradigms(approximately 2 minutes per alignment), we're interested in exploring diffeomorphic tools that are faster and can leverage GPU capabilities, which is why CLAIRE caught our attention.

Regarding our environment:
- **OS**: Ubuntu 22.04
- **Python**: 3.10
- **GPUs**: 3 x 80 GB Nvidia A100s (CUDA 11.2)
- 128 Cores (Intel) 

We would like to test all the 3 options - if possible. We have a reasonable amount of compute (multi-gpu, multi-cpu) and our goal is to have rapid results.


Kindly let me know if you need further information😃

Cheers,
Lalith

**Comment by naveenaero** [comment]
Hi @LalithShiyam ,
I will try and work on the single-GPU version of the docker image first. I will let you know when it is ready.

**Comment by LalithShiyam** [comment]
Fantastic - many thanks for your efforts! Much appreciated!

Cheers,
Lalith

**Comment by rohitrango** [comment]
I strongly recommend shipping out docker containers (or installation instructions on sandboxed/virtual environments). I tried installing the software for days, googling and debugging error messages until I hit a dead end in terms of reliable search results or lack of documentation for certain error messages.

@LalithShiyam since you use the Greedy tool, I also recommend using the [FireANTs](https://github.com/rohitrango/FireANTs) library, which has a powerful optimizer, is very easy to install on a conda environment (a one-line install), and is much faster thanks to highly optimized CUDA kernels! I have also added couple of jupyter notebook tutorials to get started.
Currently FireANTs only supports single GPU with DDP-level parallelism, but I'm working on rolling out multi-node / multi-gpu support soon.

**Comment by LalithShiyam** [comment]
Hi @rohitrango 

Fantastic! I will try your tool and keep you posted. Very excited!

And also I feel a bit relieved that I am not the only one struggling with the installation.

Cheers,
Lalith

**Comment by rohitrango** [comment]
I think the installation is a relatively common problem seeing all the open issues. 😕 But I hope the team releases docker containers for usage.

**Comment by LalithShiyam** [comment]
Hi @rohitrango,

As promised I tested the package (FireAnts), and as advertised, it is fast and delightful to work with.

If this package is in active development, I am happy to give some feedback + request some features (happy to contribute as well), which would make it way cooler than it is now.

Are there plans to continue the development? Happy to take this convo to your repo section, just to keep this thread clean.

Cheers,
Lalith

**Comment by rohitrango** [comment]
Yes. Lets take this conversation to my repo. Thanks for the feedback!
